src/languages/fsharp.js

```diff
@@ -221,16 +225,134 @@ export default function(hljs) {
+  // const OPERATOR_CHAR_RE = /[!%&*+\-\/<=>@^|~?]/;
+  // The lone dot operator is special. It cannot be redefined, and we don't want to highlight it. It can be used as part of a multi-chars operator though.
+  const OPERATOR_CHAR_OR_DOT_RE = regex.either(OPERATOR_CHAR_RE, /\./);
+  // When a dot is present, it must be followed by another operator char:
+  const OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE = regex.concat(OPERATOR_CHAR_OR_DOT_RE, regex.lookahead(OPERATOR_CHAR_OR_DOT_RE));
+  const SYMBOLIC_OPERATOR_RE = regex.either(
+    regex.concat(OPERATOR_FIRST_CHAR_OF_MULTIPLE_RE, OPERATOR_CHAR_OR_DOT_RE, '*'), // Matches at least 2 chars operators
+    regex.concat(OPERATOR_CHAR_RE, '+'), // Matches at least one char operators
+  );
+  const OPERATOR = {
+    scope: 'operator',
+    match: regex.either(
+      // symbolic operators:
+      SYMBOLIC_OPERATOR_RE,
+      // other symbolic keywords:
+      // Type casting and conversion operators:
+      /:\?>/,
+      /:\?/,
+      /:>/,
+      /:=/, // Reference cell assignment
+      /::?/, // : or ::
+      /\$/), // A single $ can be used as an operator
     relevance: 0
   };
 
+  const makeTypeAnnotationMode = function(prefix, prefixScope) {
+    return {
+      begin: regex.concat( // a type annotation is a
+        prefix,            // should be a colon or the 'of' keyword
+        regex.lookahead(   // that has to be followed by
+          regex.concat(
+            /\s*/,         // optional space
+            regex.either(  // then either of:
+              /\w/,        // word
+              /'/,         // generic type name
+              /\^/,        // generic type name
+              /#/,         // flexible type name
+              /``/,        // quoted type name
+              /\(/,        // parens type expression
+              /{\|/,       // anonymous type annotation
+      )))),
+      beginScope: prefixScope,
+      // BUG: because ending with \n is necessary for some cases, multi-line type annotations are not properly supported.
+      // Examples where \n is required at the end:
+      // - abstract member definitions in classes: abstract Property : int * string
+      // - return type annotations: let f f' = f' () : returnTypeAnnotation
+      // - record fields definitions: { A : int \n B : string }
+      end: regex.lookahead(
+        regex.either(
+          /\n/,
+          /=/)),
+      relevance: 0,
+      // we need the known types, and we need the type constraint keywords and literals. e.g.: when 'a : null
+      keywords: hljs.inherit(ALL_KEYWORDS, { type: KNOWN_TYPES }),
+      contains: [
+        COMMENT,
+        GENERIC_TYPE_SYMBOL,
+        hljs.inherit(QUOTED_IDENTIFIER, { scope: null }), // match to avoid strange patterns inside that may break the parsing
+        OPERATOR
+      ]
+    };
+  }
+
+  const TYPE_ANNOTATION = makeTypeAnnotationMode(/:/, 'operator');
+  // DU stands for Discriminated Union
```

Comment:
Lets spell it out vs hiding it in a comment. :-)  The comment doesn't exist at the reference point so someone is still left asking "what's a DU".

Reply:
Ok will do. I thought it was getting a bit too long, hence the short version.